### PR TITLE
[Camel-18635] Added ability to generate route coverage reports in HTML format

### PR DESCRIPTION
**Pull request by cziesman**
The existing version of the camel-report-maven-plugin generates route coverage statistics in a text format that, while useful, can be difficult to navigate and interpret. This PR provides the ability to generate those statistics in HTML and XLSX formats as well.

### CODE REVIEW

**Review by davsclaus — CHANGES_REQUESTED** [review]

**Comment by davsclaus** [comment]
Its a large PR - I wonder if you have considered a PR with the Excel first, and then for HTML later where we need to find a way to have a HTML page that is easy for others to maintain and don't require online access to 3rd party CSS

File: catalog/camel-report-maven-plugin/pom.xml [hunk 1]
```diff
@@ -33,6 +33,15 @@
     <name>Camel :: Catalog :: Camel Report Maven Plugin</name>
     <description>Maven plugin with tools to generate reports on your Camel projects</description>
 
+    <properties>
```
[the remark this answers is not in the document]
Comment (cziesman) [reply catalog/camel-report-maven-plugin/pom.xml:36]: Done

File: catalog/camel-report-maven-plugin/pom.xml [hunk 2]
```diff
@@ -145,12 +188,82 @@
...
+        </dependency>
+
+        <!-- test -->
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter</artifactId>
+            <version>5.9.2</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
```
[the remark this answers is not in the document]
Comment (cziesman) [reply catalog/camel-report-maven-plugin/pom.xml:200]: Done

File: catalog/camel-report-maven-plugin/pom.xml [hunk 3]
```diff
@@ -145,12 +188,82 @@
...
         </dependency>
+
     </dependencies>
 
     <build>
         <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-surefire-plugin</artifactId>
+                <version>3.1.2</version>
```
[the remark this answers is not in the document]
Comment (cziesman) [reply catalog/camel-report-maven-plugin/pom.xml:232]: Done

**Comment by cziesman** [comment]
I removed the Excel generator code. It had a few bugs anyway and can be added back later if there is a need for it. I also removed the need for online access to CSS and JS.

File: catalog/camel-report-maven-plugin/pom.xml [hunk 4]
```diff
@@ -33,6 +33,10 @@
     <name>Camel :: Catalog :: Camel Report Maven Plugin</name>
     <description>Maven plugin with tools to generate reports on your Camel projects</description>
 
+    <properties>
+        <poi-version>5.2.3</poi-version>
```
Comment (davsclaus) [inline catalog/camel-report-maven-plugin/pom.xml:37]: Can you add poi to parent pom.xml as well

**Review by davsclaus — APPROVED** [review]

File: catalog/camel-report-maven-plugin/pom.xml (hunk 4 above)
Comment (cziesman) [reply catalog/camel-report-maven-plugin/pom.xml:37]: Sorry, POI was related to Excel generation. I removed the dependencies and the property.

**Comment by davsclaus** [comment]
Thanks a lot @cziesman for all this work, you are very much welcome to keep contributing and improving this report, or other areas of Camel.

**Pull request merged**